File: OOProject1/Zookeeper.java
import java.util.*;
public class Zookeeper{
  //The animal object list is passed in for each method
  
  public static void wakeUp(Animal[] tmp){
    //Say what the Zookeeper is doing
     System.out.println("The Zookeeper is waking the animals up");
    //For each animal in the list we get their name, type, and action in that order
     for(int i = 0;i<tmp.length;i++){
        System.out.println(tmp[i].getname() + " the " + tmp[i].getClass().getName() + " " +  tmp[i].waking());
    }
    System.out.println("\n");
  }
  public static void rollCall(Animal[] tmp){
    //Say what the Zookeeper is doing
    System.out.println("The Zookeeper is doing roll call");
    //For each animal in the list we get their name, type, and action in that order
    for(int i = 0;i<tmp.length;i++){
        System.out.println(tmp[i].getname() + " the " + tmp[i].getClass().getName() + " " +  tmp[i].makeNoise());
    }
    System.out.println("\n");
  }
  public static void feed(Animal[] tmp){
    //Say what the Zookeeper is doing
    System.out.println("The Zookeeper is feeding");
    //For each animal in the list we get their name, type, and action in that order
    for(int i = 0;i<tmp.length;i++){
        System.out.println(tmp[i].getname() + " the " + tmp[i].getClass().getName() + " " +  tmp[i].eat());
    }
    System.out.println("\n");
  }
  public static void excercise(Animal[] tmp){
    //Say what the Zookeeper is doing
    System.out.println("The Zookeeper is exercising the animals");
    //For each animal in the list we get their name, type, and action in that order
    for(int i = 0;i<tmp.length;i++){
        System.out.println(tmp[i].getname() + " the " + tmp[i].getClass().getName() + " " +  tmp[i].roam());
    }
    System.out.println("\n");
  }
  public static void shutDown(Animal[] tmp){
    //Say what the Zookeeper is doing
    System.out.println("The Zookeeper is closing down the zoo");
    //For each animal in the list we get their name, type, and action in that order
    for(int i = 0;i<tmp.length;i++){
        System.out.println(tmp[i].getname() + " the " + tmp[i].getClass().getName() + " " +  tmp[i].sleep());
    }
    System.out.println("\n");
  }
}

File: Project2/Zookeeper.py
class Zookeeper(object):
    def __init__(self, zoo):
        self.observers = []
        self.zoo = zoo

    def register_observer(self, observer):
        self.observers.append(observer)

    def clear_observer(self):
        for observers in self.observers:
            del observers
        self.observers.clear()

    def notify_observers(self,action):
        for observers in self.observers:
            observers.update(action)

    def wakeUp(self):
        self.notify_observers('wake up the animals')
        for i in self.zoo:
            print(i.Name + " the " + i.Type + " " + i.wake())
        print("")

    def rollCall(self):
        self.notify_observers('roll call the animals')
        for i in self.zoo:
            print(i.Name + " the " + i.Type + " " + i.makeNoise())
        print("")

    def feed(self):
        self.notify_observers('feed the animals')
        for i in self.zoo:
            print(i.Name + " the " + i.Type + " " + i.eat())
        print("")

    def excercise(self):
        self.notify_observers('work out the animals')
        for i in self.zoo:
            print(i.Name + " the " + i.Type + " " + i.performRoam())
        print("")

    def shutDown(self):
        self.notify_observers('close the zoo')
        for i in self.zoo:
            print(i.Name + " the " + i.Type + " " + i.sleep())
        print("")
